sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1356,4 +1360,36 @@ export class ApplicationService extends RecordDataModelService<Application> {
       }
     }
   }
+
+  /**
+   * Archives one or more applications when 43 days
+   * have passed the end of the study period.
+   * @param auditUserId user making changes to table.
+   */
+  async archiveApplications(auditUserId: number): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+
+    // Build sql statement to get all application ids to archive
+    const applicationsToArchive = await this.repo
+      .createQueryBuilder("application")
+      .select("application.id")
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .where("application.applicationStatus = :completed")
+      .andWhere(
+        "(CURRENT_DATE - offering.studyEndDate) > :applicationArchiveDays",
+      )
+      .andWhere("application.isArchived <> :isApplicationArchived")
+      .getSql();
+
+    this.repo
+      .createQueryBuilder()
+      .update(Application)
+      .set({ isArchived: true, modifier: auditUser })
+      .where(`applications.id IN (${applicationsToArchive})`)
+      .setParameter("completed", ApplicationStatus.completed)
+      .setParameter("applicationArchiveDays", 43)
```

Comment:
Missed that before, please use the env variable here that is already created.

Reply:
Yeah got it.